[Lim Huan Kiat] iP

Huan-Kiat opened the pull request:
# **HUAN**
>“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/)) 

**HUAN** is a task manager that is 

* text-based
* easy to learn
* ~~FAST~~ _SUPER_ FAST to use

All you need to do is:

1. download it
2. double-click it
3. add your tasks
4. let it manage your tasks for you! 😮 

Features:

- [x] Managing tasks
- [ ] Reminders (coming soon)

If you are a Java programmer, feel free to use to for practice as well! Here's the `main` method

```java
public static void main(String[] args) {
        new Huan(FILE_PATH).run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Huan-Kiat committed:
Implement Level-0

Huan-Kiat committed:
Implement Level-1

Huan-Kiat committed:
Implement Level-2

Huan-Kiat committed:
Implement Level-3

Huan-Kiat committed:
Implement Level-4

Huan-Kiat committed:
Implement Level-5

Huan-Kiat committed:
Implement Level-6

Huan-Kiat committed:
Refactor: replace if-else with switch, introduce enums, add Javadocs

Huan-Kiat committed:
Move bulk of code from main method to run() in Huan

Huan-Kiat committed:
Merge branch 'master' into branch-Level-7

Huan-Kiat committed:
Create loadTasks method to parse file inputs

Huan-Kiat committed:
Implement Level-7

Huan-Kiat committed:
Merge branch 'branch-Level-7'

Huan-Kiat committed:
Add basic LocalDateTime functionality to Event and Deadline

Huan-Kiat committed:
Fix bugs related to datetime formatting, implement stretch goal

Huan-Kiat committed:
Merge branch 'branch-Level-8'

Huan-Kiat committed:
Implement basic OOP classes: TaskList, Ui, Storage, Parser

Huan-Kiat committed:
Implement A-MoreOOP

Huan-Kiat committed:
Merge branch 'branch-A-MoreOOP'

Huan-Kiat committed:
Implement A-Packages

Huan-Kiat committed:
Merge branch 'add-gradle-support'

Huan-Kiat committed:
Implement A-Gradle and A-JUnit

Huan-Kiat committed:
Implement A-Jar

Huan-Kiat committed:
Implement Level-9

Huan-Kiat committed:
Implement A-CodingStandard

Huan-Kiat committed:
Implement A-JavaDoc

Huan-Kiat committed:
Merge branch 'branch-Level-9'

Huan-Kiat committed:
Merge branch 'branch-A-CodingStandard'

Huan-Kiat committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/huan/ui/Ui.java

quantin96 reviewed (commented):
No coding standard related issues found
LGTM

gandwarf commented on src/main/java/huan/exception/HuanException.java:
Great job following the correct convention for tasks as a plural noun for collections. No changes needed.

gandwarf commented on src/main/java/huan/parser/Parser.java:
Extract the parsering function from the Storage, nice job

Huan-Kiat committed:
Implement A-CheckStyle

Huan-Kiat committed:
Merge branch 'branch-A-CheckStyle'

Xzh119 commented on src/main/java/huan/exception/HuanException.java:
```diff
@@ -0,0 +1,23 @@
+package huan.exception;
+
+/**
+ *  Represents a custom exception for Huan bot
+ */
+public class HuanException extends Exception {
+    /**
+     * Constructs a new HuanException with the specified message.
+     *
+     * @param description The exception message describing the error.
+     */
+    public HuanException(String description) {
+        super(description);
+    }
+
+    /**
+     * Constructs a new HuanException with no message.
+     */
+    public HuanException() {
+        super();
+    }
+
+}
```
Maybe can have more exception situations.

Xzh119 commented on src/main/java/huan/parser/Parser.java:
```diff
@@ -0,0 +1,81 @@
+package huan.parser;
+
+import huan.exception.HuanException;
+import huan.tasks.TaskList;
+import huan.ui.Huan;
+
+
+/**
+ * Handles parsing of user commands and input.
+ */
```
Comments are very clear overall.

Xzh119 commented on src/main/java/huan/storage/Storage.java:
```diff
@@ -0,0 +1,121 @@
+                String done = parts[1].trim();
+                String description = parts[2].trim();
+
+                switch (type) {
+                case "T":
+                    tasks.loadTask(new Todo(description));
+                    break;
+                case "D":
+                    if (parts.length != 4) {
+                        System.out.println("Incorrect deadline format: " + task);
+                        continue;
+                    }
+                    String by = parts[3].trim();
+                    tasks.loadTask(new Deadline(description, by));
+                    break;
+                case "E":
+                    if (parts.length != 4) {
+                        System.out.println("Incorrect event format: " + task);
+                        continue;
+                    }
```
Good use of the right indentation.

Xzh119 commented on src/main/java/huan/tasks/Task.java:
```diff
@@ -0,0 +1,55 @@
+ */
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
```
Good use of boolean naming

Xzh119 commented on src/main/java/huan/tasks/Task.java:
```diff
@@ -0,0 +1,55 @@
+/**
+ * Represents a generic task with a description and a completion status.
+ */
+public abstract class Task {
```
Good use of abstract class.

Xzh119 commented on src/main/java/huan/tasks/TaskList.java:
```diff
@@ -0,0 +1,231 @@
+     * @param id ID of task to be marked uundone.
+     * @throws HuanException for invalid IDs.
+     */
+    public void unmarkTask(int id) throws HuanException {
```
Clear naming of the method

Xzh119 reviewed (commented):
Overall, I found your code easy to read for the most part except a few places
where the nesting was too deep. I noted a few minor coding standard violations
too. Some of the classes are getting quite long. Consider splitting into
smaller classes if that makes sense.

Huan-Kiat committed:
Refactor: extract command class, remove ui package, implement Level-10

Huan-Kiat committed:
Merge branch 'branch-Level-10'

Huan-Kiat committed:
Implement A-Assertions

Huan-Kiat committed:
Refactor: remove dead code, extract variables

Removing dead code that is not used in Parser class.

Extracting return string variables to allow easier tracking of
string formats.

Huan-Kiat committed:
Merge pull request #1 from Huan-Kiat/branch-A-Assertions

Implement A-Assertions

Huan-Kiat committed:
Merge pull request #2 from Huan-Kiat/master

Merge branch 'master' into branch-A-CodeQuality

Huan-Kiat committed:
Merge branch 'master' of https://github.com/Huan-Kiat/ip

* 'master' of https://github.com/Huan-Kiat/ip:
  Implement A-Assertions

Huan-Kiat committed:
Merge branch 'branch-A-CodeQuality' of https://github.com/Huan-Kiat/ip into branch-A-CodeQuality

* 'branch-A-CodeQuality' of https://github.com/Huan-Kiat/ip:
  Implement A-Assertions

Huan-Kiat committed:
Merge branch 'master' into branch-A-CodeQuality

Huan-Kiat committed:
Merge pull request #3 from Huan-Kiat/branch-A-CodeQuality

Branch a code quality

Huan-Kiat committed:
Merge branch 'master' of https://github.com/Huan-Kiat/ip

* 'master' of https://github.com/Huan-Kiat/ip:
  Refactor: remove dead code, extract variables

Huan-Kiat committed:
Implement B-DetectAnomalies using hasConflict method

Huan-Kiat committed:
Merge branch 'branch-BCD-Extension'

Huan-Kiat committed:
Implement user guide and fix code quality issues

Huan-Kiat committed:
Re-implement JUnit tests
